service/awsconfig/v5/resource/cloudformation/update.go

```diff
@@ -64,24 +84,104 @@ func (r *Resource) newUpdateChange(ctx context.Context, obj, currentState, desir
 		return cloudformation.CreateStackInput{}, microerror.Mask(err)
 	}
 
-	r.logger.LogCtx(ctx, "debug", "finding out if the main stack should be updated")
-
-	updateState := cloudformation.UpdateStackInput{}
-
-	if currentStackState.Name != "" && !reflect.DeepEqual(desiredStackState, currentStackState) {
-		r.logger.LogCtx(ctx, "debug", "main stack should be updated")
+	if shouldScale(currentStackState, desiredStackState) {
+		r.logger.LogCtx(ctx, "debug", "main stack has to be scaled")
 
-		mainTemplate, err := r.getMainGuestTemplateBody(customObject)
+		updateState, err := r.computeUpdateState(customObject, desiredStackState)
 		if err != nil {
-			return nil, microerror.Mask(err)
+			return cloudformation.CreateStackInput{}, microerror.Mask(err)
 		}
-		updateState.StackName = aws.String(desiredStackState.Name)
-		updateState.TemplateBody = aws.String(mainTemplate)
-		// CAPABILITY_NAMED_IAM is required for updating IAM roles (worker policy).
-		updateState.Capabilities = []*string{
-			aws.String(namedIAMCapability),
+
+		return updateState, nil
+	} else {
+		r.logger.LogCtx(ctx, "debug", "main stack has not to be scaled")
+	}
+
+	// We enable/disable updates in order to enable them our test installations
+	// but disable them in production installations. That is useful until we have
+	// full confidence in updating guest clusters.
+	if updateallowedcontext.IsUpdateAllowed(ctx) {
+		r.logger.LogCtx(ctx, "debug", "finding out if the main stack has to be updated")
+
+		if shouldUpdate(currentStackState, desiredStackState) {
+			r.logger.LogCtx(ctx, "debug", "main stack has to be updated")
+
+			updateState, err := r.computeUpdateState(customObject, desiredStackState)
+			if err != nil {
+				return cloudformation.CreateStackInput{}, microerror.Mask(err)
+			}
+
+			return updateState, nil
+		} else {
+			r.logger.LogCtx(ctx, "debug", "main stack has not to be updated")
 		}
+	} else {
+		r.logger.LogCtx(ctx, "debug", "not computing update state because main stack are not allowed to be updated")
 	}
 
-	return updateState, nil
+	return cloudformation.UpdateStackInput{}, nil
+}
+
+// shouldScale determines whether the reconciled guest cluster should be scaled.
+// A guest cluster is only allowed to scale in case nothing but the worker count
+// changes. In case anything else changes as well, scaling is not allowed, since
+// any other changes should be covered by general updates, which is a separate
+// step.
+func shouldScale(currentState, desiredState StackState) bool {
+	if currentState.Name != desiredState.Name {
+		return false
+	}
```

Comment:
Just curious: why would name change?

Reply:
I have no idea. It should actually never. Here I just want to make sure nothing but the worker count changes.